crypto/compactcert/structs.go

```diff
@@ -63,23 +60,19 @@ type Reveal struct {
 	Part    basics.Participant `codec:"p"`
 }
 
-// maxReveals is a bound on allocation and on numReveals to limit log computation
-const maxReveals = 1024
-const maxProofDigests = 20 * maxReveals
-
 // Cert represents a compact certificate.
 type Cert struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
-	SigCommit    crypto.Digest   `codec:"c"`
-	SignedWeight uint64          `codec:"w"`
-	SigProofs    []crypto.Digest `codec:"S,allocbound=maxProofDigests"`
-	PartProofs   []crypto.Digest `codec:"P,allocbound=maxProofDigests"`
+	SigCommit    crypto.GenericDigest `codec:"c"`
+	SignedWeight uint64               `codec:"w"`
+	SigProofs    merklearray.Proof    `codec:"S,allocbound=MaxProofDigests"`
+	PartProofs   merklearray.Proof    `codec:"P,allocbound=MaxProofDigests"`
```

Comment:
`allocbound=MaxProofDigests` is useless, since `merklearray.Proof` is a struct and not an slice.